### PR TITLE
[node] Add `flush` option to `writeFile` in `fs/promises`

### PR DESCRIPTION
**Pull request by dislick**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/docs/latest-v20.x/api/fs.html#fspromiseswritefilefile-data-options
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/node/test/fs.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against de5b6733f17d3b631aaf79a9e23f2317a9f76402
</p>

**Review by jakebailey — DISMISSED** [review]
This can't be added until it's backported to v20, or v22 comes out. This is true for any feature added in an odd Node release.

image

**Comment by dislick** [comment]
Thanks for your feedback @jakebailey. I was under the assumption that the option has already been backported to v20:

image

The above screenshot is from the [20.x LTS docs](https://nodejs.org/docs/latest-v20.x/api/fs.html#fspromiseswritefilefile-data-options), while yours seems to be from the [21.x docs](https://nodejs.org/docs/latest-v21.x/api/fs.html#fspromiseswritefilefile-data-options) (which I mistakenly linked to in the PR description).

**Comment by jakebailey** [comment]
Ah, so it seems. Usually the docs list all versions, so not sure what's up there.

File: types/node/fs/promises.d.ts [hunk 1]
```diff
@@ -1008,6 +1008,7 @@ declare module "fs/promises" {
             | (ObjectEncodingOptions & {
                 mode?: Mode | undefined;
                 flag?: OpenMode | undefined;
+                flush?: boolean;
```
Comment (codershiba) [inline types/node/fs/promises.d.ts:1011]:
```suggestion
                /**
                  *  If all data is successfully written to the file, and 
                  * flush is `true`, `filehandle.sync()` is used to flush the data. 
                  * @default false
                  */
                flush?: boolean;
```

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
Please address feedback  from @codershiba

File: types/node/fs/promises.d.ts (hunk 1 above)
Comment (dislick) [reply types/node/fs/promises.d.ts:1016]: Thanks, I added the docs.

**Review by ghost — APPROVED** [review]

File: types/node/fs/promises.d.ts [hunk 2]
```diff
@@ -1008,6 +1008,12 @@ declare module "fs/promises" {
             | (ObjectEncodingOptions & {
                 mode?: Mode | undefined;
                 flag?: OpenMode | undefined;
+                /**
+                 * If all data is successfully written to the file, and `flush`
+                 * is `true`, `filehandle.sync()` is used to flush the data.
+                 * @default false
+                 */
+                flush?: boolean;
```
Comment (peterblazejewicz) [inline types/node/fs/promises.d.ts:1016]:
```suggestion
                flush?: boolean | undefined;
```
every other usage of flush uses undefined, same other props in this block

Comment (dislick) [reply types/node/fs/promises.d.ts]: Thanks, Peter.

**Review by ghost — APPROVED** [review]

**Review by Semigradsky — APPROVED** [review]

**Comment by Semigradsky** [comment]
Ready to merge

**Pull request merged**